include/sparrow/arrow_interface/arrow_schema.hpp

```diff
@@ -0,0 +1,143 @@
+     *              together the flag values.
+     * @param children An optional, std::vector<arrow_schema_shared_ptr>. Children pointers must not be null.
+     * @param dictionary An optional arrow_schema_shared_ptr. Must be present if the ArrowSchema represents a
+     * dictionary-encoded type. Must be nullptr otherwise.
+     * @return The created ArrowSchema unique pointer.
+     */
+    template <class F, class N, class M, class C, class D>
+        requires std::constructible_from<arrow_schema_private_data::FormatType, F>
+                 && std::constructible_from<arrow_schema_private_data::NameType, N>
+                 && std::constructible_from<arrow_schema_private_data::MetadataType, M>
+                 && std::constructible_from<arrow_schema_private_data::ChildrenType, C>
+                 && std::constructible_from<arrow_schema_private_data::DictionaryType, D>
+    [[nodiscard]] arrow_schema_unique_ptr
+    make_arrow_schema_unique_ptr(F format, N name, M metadata, std::optional<ArrowFlag> flags, C children, D dictionary);
+
+    template <class F, class N, class M, class C, class D>
+        requires std::constructible_from<arrow_schema_private_data::FormatType, F>
+                 && std::constructible_from<arrow_schema_private_data::NameType, N>
+                 && std::constructible_from<arrow_schema_private_data::MetadataType, M>
+                 && std::constructible_from<arrow_schema_private_data::ChildrenType, C>
+                 && std::constructible_from<arrow_schema_private_data::DictionaryType, D>
+    [[nodiscard]] arrow_schema_unique_ptr
+    make_arrow_schema_unique_ptr(F format, N name, M metadata, std::optional<ArrowFlag> flags, C children, D dictionary);
+
+    /**
+     * Release function to use for the `ArrowSchema.release` member.
+     */
+    void release_arrow_schema(ArrowSchema* schema);
+
+    /**
+     * Creates a default ArrowSchema unique pointer.
+     *
+     * @return The created ArrowSchema unique pointer.
+     */
+    arrow_schema_unique_ptr default_arrow_schema_unique_ptr();
+
+    void release_arrow_schema(ArrowSchema* schema)
+    {
+        SPARROW_ASSERT_FALSE(schema == nullptr)
+        SPARROW_ASSERT_TRUE(schema->release == std::addressof(release_arrow_schema))
+
+        schema->flags = 0;
+        schema->n_children = 0;
+        schema->children = nullptr;
+        schema->dictionary = nullptr;
+        schema->name = nullptr;
+        schema->format = nullptr;
+        schema->metadata = nullptr;
+        if (schema->private_data != nullptr)
+        {
+            const auto private_data = static_cast<arrow_schema_private_data*>(schema->private_data);
+            delete private_data;
+        }
+        schema->private_data = nullptr;
+        schema->release = nullptr;
+    }
+
+    inline arrow_schema_unique_ptr default_arrow_schema_unique_ptr()
+    {
+        auto ptr = arrow_schema_unique_ptr(new ArrowSchema());
```

Comment:
```suggestion
        auto ptr = arrow_schema_unique_ptr(new ArrowSchema{});
```

My understanding is that this will make the struct set to zero for members with no initializers. No need for all the lines below.